### PR TITLE
Feature/clone activity

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3393805&stafilter=NotComplete&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Services/Model/Imp/SummitService.php [hunk 1]
```diff
@@ -1817,50 +1817,134 @@ public function cloneEvent(Summit $summit, int $event_id): SummitEvent
...
 
-            if ($event->isExternalRSVP()) {
-                $eventClone->setRSVPLink($event->getRSVPLink());
+            $external_id = $event->getExternalId();
+            if ($external_id != null) {
+                $event_clone->setExternalId($external_id);
             }
 
+            $location = $event->getLocation();
+            if ($location != null) {
```
Comment (smarcet) [inline app/Services/Model/Imp/SummitService.php:1841]: use hasLocation here

File: app/Services/Model/Imp/SummitService.php [hunk 2]
```diff
@@ -1817,50 +1817,134 @@ public function cloneEvent(Summit $summit, int $event_id): SummitEvent
...
+                $event_clone->setLocation($location);
+            }
+
+            $local_start_date = $event->getLocalStartDate();
+            if ($local_start_date != null) {
+                $event_clone->setStartDate($local_start_date);
+            }
+
+            $local_end_date = $event->getLocalEndDate();
```
Comment (smarcet) [inline app/Services/Model/Imp/SummitService.php:1850]: this is not correct
we dont use local internally

File: app/Services/Model/Imp/SummitService.php [hunk 3]
```diff
@@ -1817,50 +1817,134 @@ public function cloneEvent(Summit $summit, int $event_id): SummitEvent
...
 
             // check if SummitEventWithFile
 
             if ($event instanceof SummitEventWithFile && $event->hasAttachment()) {
-                $eventClone->setAttachment($event->getAttachment());
+                $event_clone->setAttachment($event->getAttachment());
             }
 
-            $this->event_repository->add($eventClone);
+            if ($event instanceof Presentation) {
```
Comment (smarcet) [inline app/Services/Model/Imp/SummitService.php:1911]: also need special cases fro following derived classes
* SummitGroupEvent
* SummitEventWithFile

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Services/Model/Imp/SummitService.php (hunk 3 above)
Comment (romanetar) [reply app/Services/Model/Imp/SummitService.php:1921]: SummitEventWithFile already had special treatment (https://github.com/OpenStackweb/summit-api/pull/199/files#diff-a4e3ced07cbae056279455d51c16cb0ce5d06fa73af7c022b5c40fac42af10bfR1907)

I added the SummitGroupEvent part

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**